src/crypto/qat/qcccrypto.h

```diff
@@ -6,9 +6,21 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <pthread.h>
+#include <thread>
+#include <mutex>
 #include <queue>
+#include <memory>
+#include "common/async/yield_context.h"
+#include "common/async/completion.h"
```

Comment:
it doesn't look like `completion.h` is used, though it could potentially be used instead of `std::function`

however, we're planning to upgrade to boost 1.81 soon which adds a `boost::asio::any_completion_handler` that i hope will replace our existing uses of `ceph::async::Completion`. boost is missing docs for `any_completion_handler`, but there's an example under 'Type Erasure':
https://www.boost.org/doc/libs/1_81_0/doc/html/boost_asio/examples/cpp20_examples.html#boost_asio.examples.cpp20_examples.type_erasure

for now i think `std::function` is probably fine